test/extended/util/disruption/controlplane/controlplane.go

```diff
@@ -162,3 +163,38 @@ func allClusterVersionsAreGTE(version semver.Version, config *rest.Config) (bool
 	}
 	return true, nil
 }
+
+type ApiFastRestorationTest struct{}
+
+func (t ApiFastRestorationTest) Name() string { return "api-fast-restoration-test" }
+func (t ApiFastRestorationTest) DisplayName() string {
+	return "API is being restored within restricted time"
+}
+
+// Setup does nothing
+func (t *ApiFastRestorationTest) Setup(f *framework.Framework) {
+}
+
+// Test runs a connectivity check to the core APIs.
```

Comment:
why should we check a connectivity invariant while rolling an kube-apiserver revision out that causes 60s of outage?

Reply:
refactored the code to make it a lot simpler, please have another look